pymongo/collection.py

```diff
@@ -608,9 +608,9 @@ def _update(self, sock_info, criteria, document, upsert=False,
             else:
                 update_doc['arrayFilters'] = array_filters
         if hint is not None:
-            if not acknowledged:
+            if sock_info.max_wire_version < 8:
```

Comment:
A strict reading of the spec (and spec tests) actually requires this to be: `if not acknowledged && sock_info.max_wire_version < 8`:
```
class UpdateOptions {
...
   * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
   * This option is only supported by servers >= 4.2. Older servers >= 3.4 will report an error for using this option.
   * For servers < 3.4, the driver MUST raise an error if the caller explicitly provides a value.
   * For unacknowledged writes using OP_UPDATE, the driver MUST raise an error if the caller explicitly provides a value.
   * For unacknowledged writes using OP_MSG and servers < 4.2, the driver MUST raise an error if the caller explicitly provides a value.
   *
   * @see https://docs.mongodb.com/manual/reference/command/update/
   */
  hint: Optional<(String | Document)>;
```
https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#insert-update-replace-delete-and-bulk-writes

I think this is why the test are failing.

Reply:
Fixed